lectures/06_lecture/app.ipynb

```diff
@@ -0,0 +1,304 @@
+    "If everything goes according to the plan, we should be able to visit three endpoints.\n",
+    "\n",
+    "- [http://127.0.0.1:8000/](http://127.0.0.1:8000/) - root endpoint with \"Hello world\" example data.\n",
+    "- [http://127.0.0.1:8000/items/1?q=hello_again](http://127.0.0.1:8000/items/1?q=hello_again) - endpoint with example of using both path and query parameters.\n",
+    "- [http://127.0.0.1:8000/docs](http://127.0.0.1:8000/docs) - automatically generated documentation.\n",
+    "\n",
+    "The main adavantages of FastAPI framework are following:\n",
+    "\n",
+    "- high performance,\n",
+    "- development time reduction due to intuitive API,\n",
+    "- automatically generated documentation."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8dcb50e9",
+   "metadata": {},
+   "source": [
+    "### Edge AI\n",
+    "\n",
+    "Let's get back to our example and remind ourselves how the project has evolved so far. In the first phase, client \n",
+    "has recieved the web application for testing purposes. When client was happy with the results, we have created an\n",
+    "API enpoint with the model that returns model predictions, based on the input. The client implemented the solution\n",
+    "to their internal system.\n",
+    "\n",
+    "However, imagine the following situation. The client wants to use the inference also on mobile devices in their\n",
+    "own mobile app. Basically, there are two options.\n",
+    "\n",
+    "1. Mobile device can send the requests to the internal system (backend) where the solution is already prepared.\n",
+    "1. There can be model installed with the app, and it can be saved into the mobile device itself.\n",
+    "\n",
+    "The first option is almost ready. The mobile developers will only implement the inference part to their code. \n",
+    "However, from the user perspective, the second option is much more convenient. If model sits in the device, it \n",
+    "can improve an inference latency a lot, the user doesn't have to rely on the internet connection, so the \n",
+    "predictions can be used in offline mode, and, last but not least, it is much safer to keep the data on the device\n",
+    "instead of sending it somewhere else.\n",
+    "\n",
+    "However, implementing models on the mobile devices is not for free for us, ML Engineers. We have to create a\n",
+    "model that is small and preferably platform agnostic, which, in some cases, is not so easy. Also, wee need \n",
+    "to accept the fact that by reducing model size, we probbaly loose some of the model accuracy. And at the end, we \n",
+    "need to cooperate closely with mobile developers to make sure that model is used and interpreted correctly.\n",
+    "\n",
+    "As we can see, the deployment of ML model on mobile devices, resp. edge AI, is much different from the previous\n",
+    "use cases. But it is not impossible. Big players on market, such as [Tensorflow](https://www.tensorflow.org/) \n",
```

Comment:
Instead of links to TF and PyTorch I would mention that there is TFLite for edge deployment with a link like this: https://www.tensorflow.org/lite and for PyTorch there is PyTorch Mobile: https://pytorch.org/mobile/home/ 
You can also write something about the fact that the whole pipeline is not changed, you still use TF/PyTorch for model training but in the end, you just convert a model to TFLite or PyTorch mobile or CoreML or ONNX and use an interpreter for selected format on iOS or Android